### DISCUSSION TITLE
All players are disconnected from the server at the same time

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15076] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Suddenly server kicks all players

### Reproduction steps

play with 16+ players on heavy modded server where many events will happen

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
[11/06/2024 21:00:55]
  Erthex was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  Erthex已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  小散伞 was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  小散伞已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  疯狂星期四 was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  疯狂星期四已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  Mark was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  Mark已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  可口可乐无糖香草味 was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  可口可乐无糖香草味已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  苦艾酒 was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  苦艾酒已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  guizuns was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  guizuns已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  屑狗子 was kicked because they were expecting a very old network event (46552 (created 211.52 s ago, 211.42 s older than last event sent to anyone) Events queued: 32810, last sent to all: 13772
[11/06/2024 21:00:55]
  屑狗子已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:55]
  阿莱斯特 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:55]
  阿莱斯特已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:55]
  Erthex was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:55]
  Erthex已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:55]
  小散伞 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:55]
  小散伞已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:55]
  suger 阿糖 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  suger 阿糖已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  机械蜈蚣 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  机械蜈蚣已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  疯狂星期四 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  疯狂星期四已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  Mark was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  Mark已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  凛冬眠 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  凛冬眠已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  可口可乐无糖香草味 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  可口可乐无糖香草味已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  苦艾酒 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  苦艾酒已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  guizuns was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  guizuns已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  屑狗子 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  屑狗子已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  9Eagle has gained 316 experience (37637 -> 37953)
[11/06/2024 21:00:56]
  Yukikaze was kicked because they were expecting a very old network event (46552 (created 211.53 s ago, 211.42 s older than last event sent to anyone) Events queued: 32813, last sent to all: 13772
[11/06/2024 21:00:56]
  Yukikaze已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:56]
  四月鱼 was kicked because they were expecting a very old network event (46552 (created 211.53 s ago, 211.42 s older than last event sent to anyone) Events queued: 32813, last sent to all: 13772
[11/06/2024 21:00:56]
  四月鱼已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:56]
  rickrick142 was kicked because they were expecting a very old network event (46552 (created 211.53 s ago, 211.42 s older than last event sent to anyone) Events queued: 32813, last sent to all: 13772
[11/06/2024 21:00:56]
  rickrick142已断开连接。理由: 事件失去同步(正等待旧事件)
[11/06/2024 21:00:56]
  Yukikaze was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  Yukikaze已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  四月鱼 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  四月鱼已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  rickrick142 was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  rickrick142已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  9Eagle was kicked because they were expecting a removed network event (46552, last available is 13772)
[11/06/2024 21:00:56]
  9Eagle已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  2112GJR was kicked because they were expecting a removed network event (46602, last available is 13772)
[11/06/2024 21:00:56]
  2112GJR已断开连接。理由: 事件失去同步(正等待已移除事件)
[11/06/2024 21:00:56]
  服务器内全部玩家已死亡，在本巡回内不能复活。如果没有玩家复活或加入，服务器将在120秒后结束这一巡回。
```

**Comment by A2ur3** [comment]
#13458 was the issue when we did not play on dedicated server

**Comment by A2ur3** [comment]
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is coilgunbolt, client entity is huskmutanthumanheadhijackparticle). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is flakshrapneldirectional, client entity is flakboltdirectional). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is flakshrapneldirectional, client entity is flakboltdirectional). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is flakbolt, client entity is coilgunbolt). Ignoring the message...
WARNING: Attempted to play the sound Barotrauma.Sounds.OggSound (D:/Steam/steamapps/common/Barotrauma/Content/Characters/Fractalguardian/FRACTALGUARDIAN2_attack1.ogg) while it was still loading.
WARNING: Attempted to play the sound Barotrauma.Sounds.OggSound (D:/Steam/steamapps/common/Barotrauma/Content/Characters/Fractalguardian/FRACTALGUARDIAN1_attack1.ogg) while it was still loading.
WARNING: Attempted to play the sound Barotrauma.Sounds.OggSound (D:/Steam/steamapps/common/Barotrauma/Content/Characters/Fractalguardian/FRACTALGUARDIAN1_attack3.ogg) while it was still loading.
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_58, client entity is tsm_9_ammo). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message. Entity type does not match (server entity is not an item, client entity is Barotrauma.Item). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message. Entity type does not match (server entity is not an item, client entity is Barotrauma.Item). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_9_ammo, client entity is plastic). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_9_ammo, client entity is coilgunbolt). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_9_ammo, client entity is tsm_58). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is scp_762x39duprojectile, client entity is tsm_9_ammo). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_9_ammo, client entity is scp_762x39duprojectile). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is scp_762x39duprojectile, client entity is tsm_9_ammo). Ignoring the message...
WARNING: Received a potentially invalid ENTITY_POSITION message.Entity identifier does not match (server entity is tsm_9_ammo, client entity is screwdriver). Ignoring the message...

[12/15/2024 21:06:04]
  WARNING: ServerEntityEventManager is lagging behind! Last sent id: 31162, latest create id: 64058
1. EntitySpawner x6607
2. Fractalguardian3 x1817
3. 保险钢柜（医生） (ID: 2755) x1469 (content package: EK Utilities 重製版)
[12/15/2024 21:06:04]
  WARNING: ServerEntityEventManager is lagging behind! Last sent id: 31162, latest create id: 64081
1. EntitySpawner x6607
2. Fractalguardian3 x1817
3. 保险钢柜（医生） (ID: 2755) x1469 (content package: EK Utilities 重製版)

**Reply by Regalis11** [reply]
> EntitySpawner x6607

This seems very strange to me, it would imply that something is causing an enormous amount of entities to be spawned. I wonder if some mod could be causing this?

**Reply by A2ur3** [reply]
at that moment 5 players were fighting fractal guardians and 4 were keeping firing their weapons
and server is running a lua script that updates EntitySpawner to prevent issues caused by Performance Fix
```lua
Hook.Patch("Barotrauma.MapEntity", "UpdateAll", function()
    if Entity.Spawner then
        Entity.Spawner.Update()
    end
end, Hook.HookMethodType.After)
```
but the collective disconnecting is likely not caused by the script. It happened before when the script is not run
and it may even isn't related to EntitySpawner because the disconnecting happened yesterday when we docked on outpost where nothing strange was happening

**Reply by A2ur3** [reply]
does removing character or item increase EntitySpawner event counts? If it does then it makes sense why EntitySpawner event counts are very high because we were probably despawning many corpses and items inside their inventories

**Comment by A2ur3** [comment]
Coroutine LevelTransition threw an exception: Object reference not set to an instance of an object.
   at Barotrauma.MultiPlayerCampaign.DoLevelTransition()+MoveNext() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameSession\GameModes\MultiPlayerCampaign.cs:line 315
   at Barotrauma.CoroutineManager.PerformCoroutineStep(CoroutineHandle handle) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\CoroutineManager.cs:line 214
   at Barotrauma.CoroutineManager.IsDone(CoroutineHandle handle) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\CoroutineManager.cs:line 236

**Comment by A2ur3** [comment]
Entity: Hydroxyapatite (CaP) (ID: 11690)
Exception thrown while reading a message of the type "EntityEvent" at position 32.
Inner exception: Object reference not set to an instance of an object.
   at Barotrauma.Items.Components.LevelResource.ClientEventRead(IReadMessage msg, Single sendingTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Components\LevelResource.cs:line 15
   at Barotrauma.Item.ClientEventRead(IReadMessage msg, Single sendingTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Item.cs:line 1957
   at Barotrauma.Networking.ClientEntityEventManager.Read(ServerNetSegment type, IReadMessage msg, Single sendingTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\NetEntityEvent\ClientEntityEventManager.cs:line 235

**Comment by rekterakathom** [comment]
I'd like to chime in and report problems with the network events as well. Ever since the winter update, it has been very common for players to be kicked for old events.

``player was kicked because they were expecting a very old network event (2919 (created 10.12 s ago, 10.03 s older than last event sent to anyone) Events queued: 393, last sent to all: 2918``

**Comment by 4-S-H** [comment]
It may be due to a lot of events related to hull being suddenly created
https://github.com/FakeFishGames/Barotrauma/discussions/15628

**Comment by A2ur3** [comment]
can confirm the issue comes from #15625 (#15436)
When a client can't sync but still joins and gets stuck, the server keeps trying to sync it (and fails)
eventually server disconnects everyone
